File: api-tester/src/defaults/java/com/revenuecat/apitester/java/revenuecatui/PaywallResultAPI.java
package com.revenuecat.apitester.java.revenuecatui;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.activity.result.ActivityResultCallback;
import androidx.annotation.NonNull;

import com.revenuecat.purchases.CustomerInfo;
import com.revenuecat.purchases.PurchasesError;
import com.revenuecat.purchases.ui.revenuecatui.ExperimentalPreviewRevenueCatUIPurchasesAPI;
import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallResult;
import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallResultHandler;

@SuppressWarnings({"unused"})
@ExperimentalPreviewRevenueCatUIPurchasesAPI
final class PaywallResultAPI {

    static void checkResultHandler(
            final PaywallResultHandler resultHandler
    ) {
        final ActivityResultCallback<PaywallResult> callback = resultHandler;
    }

    static void checkResult(PurchasesError error, CustomerInfo customerInfo) {
        final PaywallResult result = PaywallResult.Cancelled.INSTANCE;
        final Parcelable parcelable = result;
        final PaywallResult.Error result2 = new PaywallResult.Error(error);
        PurchasesError error2 = result2.getError();
        final PaywallResult.Purchased result3 = new PaywallResult.Purchased(customerInfo);
        CustomerInfo customerInfo2 = result3.getCustomerInfo();
        final PaywallResult result2AsSealedClass = result2;
        final PaywallResult result3AsSealedClass = result3;
    }
}

File: api-tester/src/defaults/kotlin/com/revenuecat/apitester/kotlin/revenuecatui/PaywallResultAPI.kt
package com.revenuecat.apitester.kotlin.revenuecatui

import android.os.Parcelable
import androidx.activity.result.ActivityResultCallback
import com.revenuecat.purchases.CustomerInfo
import com.revenuecat.purchases.PurchasesError
import com.revenuecat.purchases.ui.revenuecatui.ExperimentalPreviewRevenueCatUIPurchasesAPI
import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallResult
import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallResultHandler

@Suppress("unused", "UNUSED_VARIABLE")
@OptIn(ExperimentalPreviewRevenueCatUIPurchasesAPI::class)
private class PaywallResultAPI {
    fun checkResultHandler(
        resultHandler: PaywallResultHandler,
    ) {
        val callback: ActivityResultCallback<PaywallResult> = resultHandler
    }

    fun checkResult(paywallResult: PaywallResult, error: PurchasesError, customerInfo: CustomerInfo) {
        when (paywallResult) {
            is PaywallResult.Cancelled -> {}
            is PaywallResult.Error -> {
                val error2: PurchasesError = paywallResult.error
            }
            is PaywallResult.Purchased -> {
                val customerInfo2: CustomerInfo = paywallResult.customerInfo
            }
            is PaywallResult.Restored -> {
                val customerInfo3: CustomerInfo = paywallResult.customerInfo
            }
        }
        val parcelable: Parcelable = paywallResult
        val result2 = PaywallResult.Error(error)
        val result3 = PaywallResult.Purchased(customerInfo)
        val result4 = PaywallResult.Restored(customerInfo)
    }
}
